web/source/kmwutils.ts

```diff
@@ -951,22 +951,20 @@ namespace com.keyman {
     }
 
     /**
-     * Customized alert
+     * Customized alert. This is enabled/disabled by the option flag 'useAlerts'
      *
      * @param     {string}        s       alert text
      * @param     {function()=}   fn      function to call when alert dismissed
      */
     alert(s: string, fn?: () => void): void {
```

Comment:
### Required change

- [ ] We should be conditioning _internal_ use of `alert` on the new option, not the method's functionality.

Yes, this means finding all (affected) uses of the `alert` function and conditioning those, rather than the 'simple' change here.

### Reasoning

Please note that `keyman.util.alert` is one of [our published API functions](https://help.keyman.com/DEVELOPER/engine/web/14.0/reference/util/alert).  As such, a site user may decide to _intentionally_ use this method, and this would break such uses if they set `useAlerts` to `false`.

After all, a user might decide to handle alerts themselves so that they might customize (or maybe even localize) the alert message for use on their page.

Reply:
> Yes, this means finding all (affected) uses of the alert function and conditioning those, rather than the 'simple' change here.

`useAlerts` is an all-or-nothing flag. I don't see users could selectively choose for some internal alerts to display and not others.

`useAlerts` also defaults to true, so the published API still behaves the same. If they intentionally set `useAlerts` to `false`, then  surely they are handling alerts themselves.
Granted the 15.0 API would be updated to mention the new `useAlerts` flag.

Reply:
I see no reason to intentionally cripple the ability for a site designer to use it with `useAlerts` off.  It's very simple to just turn off KMW's direct uses of it while still allowing site designers to utilize it.

Reply:
Please do this as a wrapper function for `alert()` rather than testing everywhere.

Reply:
Added a wrapper `util.internalAlert()`